### DISCUSSION TITLE
Add GraphQL support

### DISCUSSION THREAD

**Discussion by seladb** [litestar-org/litestar#605] [Ideas]
GraphQL is becoming very popular. Do we want to support it in Starlite?

It's supported in FastAPI: https://fastapi.tiangolo.com/advanced/graphql/
And in Starlette through 3rd-party libraries: https://www.starlette.io/graphql/

We probably want to decide which GraphQL library to integrate first, maybe [Strawberry](https://strawberry.rocks/)?

**Comment by Goldziher** [comment]
Yes strawberry us a natural candidate here. What do we need to support this?

**Reply by seladb** [reply]
That's a good question, I haven't checked yet. They already support Starlette so I guess not too much, but since Starlite doesn't expose Starlette directly in most cases, there might be some work there (either on Starlite and/or on the Strawberry side). 

If you're open to the idea I can test it and let you know.

**Reply by Goldziher** [reply]
Sure

**Reply by dinkopehar** [reply]
Strawberry has experimental support with pydantic. It would be nice if it would integrate with Strawberry in some way because lot of frameworks are supported by Strawberry.

**Comment by teddybee** [comment]
Any progress about this? I would like to start a project with Litestar, but some day I may need graphql (with subscriptions) in the api. I feel the decision risky.

**Reply by provinzkraut** [reply]
I’m still not really sure what an integration would look like. Cooks you explain a bit what you’re looking for?

If it’s just using e.g. strawberry like suggested in this discussion originally, you can already just do that, there’s no need for any special support since it’s just mounting an ASGI app.

**Reply by teddybee** [reply]
Maybe an example app(with integrated input validation) or a suggestion would be useful in the docs. 
Thank you, I will try it soon.

**Reply by JacobCoffee** [reply]
https://github.com/litestar-org/litestar/issues/3317

**Comment by ryou90** [comment]
https://strawberry.rocks/docs/integrations/litestar

Strawberry already supports Litestar :)

**Comment by JacobCoffee** [comment]
This is available via Strawberry and we have an open issue to add docs.
